osu.Game/Rulesets/Scoring/DrainingHealthProcessor.cs

```diff
@@ -86,6 +97,7 @@ protected override void ApplyResultInternal(JudgementResult result)
         {
             base.ApplyResultInternal(result);
             healthIncreases.Add((result.HitObject.GetEndTime() + result.TimeOffset, GetHealthIncreaseFor(result)));
+            applyDrain = true;
```

Comment:
A similar change needs to be done to `computeDrainRate()` below (change probably needs to happen to line 144), which simulates this whole `Update()` + `ApplyResult` cycle.